src/main/flight/imu.c

```diff
@@ -403,24 +403,39 @@ static void imuMahonyAHRSupdate(float dt, const fpVector3_t * gyroBF, const fpVe
         if (velEF && vVelEF_initialized) {
             // We have valid velocity from GPS, use that to calculate correction vector
             fpVector3_t vAccEF;
+            vectorZero(&vAccEF);
 
             // Rotate ACC from BF to EF and accumulate
             quaternionRotateVectorInv(&vAccEF, accBF, &orientation);
+
+            vectorScale(&vAccEF, &vAccEF, 1.0f/980.0f); //scale g
+            vectorScale(&vAccEF, &vAccEF, dt); //integrate 
+
             vectorAdd(&vVelEF_AccIntegal, &vVelEF_AccIntegal, &vAccEF);
             vVelEF_integralTime += dt;
 
             // Now if we got a GPS update we can calculate error vector according to
             // the paper by Bill Premerlani (Roll-Pitch Gyro Drift Compensation)
             if (velEFNew) {
                 fpVector3_t vTmp1, vTmp2;
+                vectorZero(&vTmp1);
+                vectorZero(&vTmp2);
 
                 // Calculate (V2 - V1) / (T2 - T1)
                 // Here T2-T1 = vVelEF_integralTime (accumulated time between two GPS updates, also time window we used to integrate accelerometer)
-                vectorSub(&vTmp1, &vVelEF_prev, velEF);
+             
+                fpVector3_t velEFLocal;
+
+                vectorScale(&velEFLocal, velEF, 1.0f);
+
+                vectorScale(&velEFLocal, &velEFLocal, 1.0f / 980.0f); //scale EF to match g scale
```

Comment:
I don't think we should scale to G's, cm/s/s should be enough as long as units are the same for GPS-based acc and IMU-based